### DISCUSSION TITLE
How to use workspaces?

### DISCUSSION THREAD

**Discussion by opsb** [vitest-dev/vitest#3214] [Q&A]
I've been following the instructions at https://vitest.dev/guide/workspace.html#configuration and have the following packages in the root of a monorepo: (each contains a `vite.config.ts`)

```
packages/core
packages/frontend
```

In`vitest.workspace.json` I simply have `["packages/*"]`. 

When I `npm run test` (defined as just `vitest`) I'm shown a list of all the test files with a `(0)` next to them and lots of errors like:
```
Error: No test suite found in file /Users/user123/Projects/myproject/packages/core/src/utils/tree.test.ts
```
Is this configuration expected to work?

vitest version: 0.30.1

**Comment by opsb** [comment]
I'll have to chalk that one up to late night coding, helps if you upgrade vitest in the packages as well... Leaving here in case anyone happens to make the same mistake.

**Reply by sheremet-va** [reply]
You shouldn't probably have a vitest installed in your project `package.json`'s

**Reply by opsb** [reply]
Hey, just to clarify, you mean vitest should only be in the root package.json when using workspaces?

**Reply by sheremet-va** [reply]
Yes

**Reply by opsb** [reply]
I'm new to workspaces (switching a project over to pnpm workspaces at the moment), I hadn't realised dependencies installed in the root would be shared with all packages, thanks for the pointer.

**Comment by opsb** [comment]
One little wrinkle I ran into, tailwind wasn't resolving its config correctly. Originally I had

```javascript
import tailwindcss from "tailwindcss";
import autoprefixer from "autoprefixer";

const config = {
  plugins: [tailwindcss("./tailwind.config.ts"), autoprefixer()],
};

export default config;
```

and tailwind was throwing an error because it's config was undefined. Importing the config directly fixed it.

```javascript
import tailwindcss from "tailwindcss";
import autoprefixer from "autoprefixer";
import tailwindConfig from "./tailwind.config";

const config = {
  plugins: [tailwindcss(tailwindConfig), autoprefixer()],
};

export default config;
```
